lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -149,6 +174,77 @@ def remove_schedule(name)
 
       private
 
+      def supports_lua?
+        redis_master_version >= 2.5
+      end
+
+      def redis_master_version
+        Resque.redis.info['redis_version'].to_f
+      end
+
+      def startup_sha(refresh = false)
+        @startup_sha = nil if refresh
+
+        @startup_sha ||=
+          Resque.redis.script(:load, <<-EOF.gsub(/^ {14}/, ''))
+            local schedules_key = KEYS[1]
+            local persisted_schedules_key = KEYS[2]
+            local changed_schedules_key = KEYS[3]
+
+            local remove_schedule = function(schedule)
+              redis.call("HDEL", schedules_key, schedule)
+              redis.call("SREM", persisted_schedules_key, schedule)
+              redis.call("SADD", changed_schedules_key, schedule)
+            end
+
+            local get_non_persistent_schedules = function()
+              local all_schedules = redis.call("HKEYS", schedules_key)
+              local non_persistent_schedules = {}
+              for _,schedule in ipairs(all_schedules) do
+                local schedule_is_persisted = \
+                  redis.call("SISMEMBER", persisted_schedules_key, schedule)
+                if schedule_is_persisted == 0 then
+                  table.insert(non_persistent_schedules, schedule)
+                end
+              end
+              return non_persistent_schedules
+            end
+
+            local set_schedule = function(name, config, persisted)
+              redis.call("HSET", schedules_key, name, config)
+              redis.call("SADD", changed_schedules_key, name)
```

Comment:
It's how dynamic schedules are updated. https://github.com/Shopify/resque-scheduler/blob/master/lib/resque/scheduler.rb#L298-L299